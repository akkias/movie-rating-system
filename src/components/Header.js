import React, {useState} from 'react';
import {getRandomNum} from '../utils'

export const Header = (props) => {
    let setTM;
    const [randomRating, setRR] = useState(false);

    async function setRandomRating() {
        try {
            setRR(true);
            let min = 5, max = 10;
            var myFunction = function() {
                props.setRating(getRandomNum(5),getRandomNum(12), getRandomNum(12));
                let randomTime = Math.round(Math.random() * (max - min + 1) + min);
                console.log(randomTime)
                setTM = setTimeout(myFunction, randomTime * 1000);
            }
            await myFunction();
            console.log('here')
        }
        catch (err) {
            console.log('fetch failed', err);
        }
    } 
    const stopRandomRating = () => {
        clearTimeout(setTM);
        setRR(false);
    }
    return(
        <div className="header">
            <h1>Movie Rating App</h1>
            {!randomRating ? 
            <button className="random-rating-btn" onClick={setRandomRating}>
                <i className="fas fa-star"></i>Random Rating
            </button>
            :
            <button className="random-rating-btn" onClick={stopRandomRating}>
                <i className="fas fa-sync fa-spin"></i>Stop Random Rating
            </button>
            }
        </div>
    )
}

export default Header;