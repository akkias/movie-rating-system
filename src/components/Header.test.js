import React from 'react';
import { mount } from 'enzyme';
import Header from './Header'


describe('<Header />', () => {
    const header = mount(<Header />);
    it('<Header /> Renders correctly', () => {
        expect(<header />).toMatchSnapshot();
    })
  
})
