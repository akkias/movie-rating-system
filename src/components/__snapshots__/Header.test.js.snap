// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> <Header /> Renders correctly 1`] = `<header />`;
